src/agent0/hyperfuzz/system_fuzz/invariant_checks.py

```diff
@@ -390,46 +388,59 @@ def _check_lp_share_price(
     exception_data: dict[str, Any] = {}
     log_level = None
 
-    block_number = pool_state.block_number
+    # TODO we hack in a stateful variable into the interface here, since we need
```

Comment:
you could add the variable as a function arg and include it in the InvariantCheckResults, so that it maintains state across subsequent calls